### ISSUE TITLE
Navigator causing problems when used with ImagePicker

### ISSUE THREAD

**Issue by lschmierer** [flutter/flutter#13600]
## Steps to Reproduce

When a Fullscreen dialog is opened after a file was received from `ImagePicker`, the layout of the Fullscreen Dialog is broken.
```Dart
final File _ = await ImagePicker.pickImage();

Navigator
    .of(context)
    .push(new MaterialPageRoute<Null>(
  builder: (final BuildContext context) {
    return new TestPage();
  },
));
```
![device-2017-12-15-105913](https://user-images.githubusercontent.com/2204708/34037290-8c7f1c84-e188-11e7-8301-7b88fb382d69.png)

However, wrapping the call to `Navigator.push` into a new `Future` fixes the issue.
```Dart
final File _ = await ImagePicker.pickImage();

new Future(() {
  Navigator
      .of(context)
      .push(new MaterialPageRoute<Null>(
    builder: (final BuildContext context) {
      return new TestPage();
    },
  ));
});
```

See [this](https://github.com/lschmierer/flutter_test) project if you want to reproduce the issue.

## Flutter Doctor
```
[✓] Flutter (on Mac OS X 10.13.1 17B1003, locale en-DE, channel master)
    • Flutter at /Users/lschmierer/flutter
    • Framework revision 6299dbada8 (13 hours ago), 2017-12-14 13:19:26 -0800
    • Engine revision e0d19e2897
    • Tools Dart version 1.25.0-dev.11.0
    • Engine Dart version 2.0.0-edge.cf1de7d46cd88e204380e8f96a993439be56b24c

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.1)
    • Android SDK at /Users/lschmierer/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    • ios-deploy 1.9.2
    • CocoaPods version 1.3.1

[✓] Android Studio (version 3.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] Connected devices
    • Pixel • FA69N0312108 • android-arm • Android 8.1.0 (API 27)
```

**Comment by lschmierer** [comment]
On the iOS simulator it does not work at all, even with the `Future` workaround,

**Comment by sthai256** [comment]
Having the same issue as well when I try to present a confirmation dialog after the picker.

**Comment by lschmierer** [comment]
This issue should probably closed
Much more information over at https://github.com/flutter/flutter/issues/13818

**Comment by mravn-google** [comment]
Closed as duplicate of #13818
